src/commonMain/kotlin/main.kt

```diff
@@ -568,3 +570,26 @@ private fun help(exeFile: File?, programName: String, supportedOptions: JaunchOp
     val optionsUnique = linkedSetOf(*supportedOptions.values.toTypedArray())
     optionsUnique.forEach { printlnErr(it.help()) }
 }
+
+private fun applyUpdate(appDir: File, updateSubDir: File) {
+    if (!updateSubDir.exists) return
+
+    // Recursively copy over all files in the update subdir
+    for (file in updateSubDir.ls()) {
+        if (file.isDirectory) applyUpdate(appDir, file)
+        else {
+            val dest = appDir / file.path.substring((appDir / "update").path.length)
+            if (file.stat() == 0L) {
+                val rmDest = dest.rm()
+                if (!rmDest) error("Couldn't remove $dest")
```

Comment:
One concise pattern in Kotlin is the Elvis operator `?:` which triggers when the left side evaluates to null. So if we make `rm()` and `mv` and `rmdir` return null upon failure, you could then write:
```kotlin
dest.rm() ?: error("Couldn't remove $dest")
```
which would read nicely. What do you think?

Reply:
Alternately, I think the following syntax might work:
```kotlin
dest.rm() || error("Couldn't remove $dest")
```